unikernel.ml

```diff
@@ -66,6 +101,24 @@ struct
       ~on_eof:(on_eof f_init);
     finished >>= fun data -> data
 
+  let extract_csrf_token reqd =
+    decode_request_body reqd >>= fun data ->
+    let json =
+      try Ok (Yojson.Basic.from_string data)
+      with Yojson.Json_error s -> Error (`Msg s)
+    in
+    match json with
+    | Error (`Msg err) ->
+        Logs.warn (fun m -> m "Failed to parse JSON: %s" err);
+        Lwt.return (String.empty, `Null)
```

Comment:
I don't quite understand why using `String.empty` here? Wouldn't a `Error ()` or a `None` be more in the OCaml spirit?

Reply:
done in b4aee8c